### ISSUE TITLE
Some lines seem to be unreachable code.

### ISSUE THREAD

**Issue by yangby-cryptape** [nervosnetwork/merkle-mountain-range#34]
I have run fuzz tests again and again, the following `if` condition is never entered:

https://github.com/nervosnetwork/merkle-mountain-range/blob/8206b586c3534cce2a4ddbd0ffbc5e2a9ce04727/src/mmr.rs#L404-L406

I could NOT prove but I think it is unreachable.

**Comment by quake** [comment]
it's reachable, I added an unit test: https://github.com/nervosnetwork/merkle-mountain-range/pull/36

**Comment by quake** [comment]
Yes, better to return `CorruptedProof` directly, and I noticed that line 408 checked the iterator empty again, so these lines can be deleted directly, cc @jjyr
